cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -116,16 +122,31 @@ private void addPrefixes(@Nonnull Map<String, String> prefixes,
   }
 
   private void addTypes(Configuration configuration, RdfStoreService rdfStoreService, Model shacl) {
-    List<String> types = getTypes(rdfStoreService);
+    List<String> types = getTypes(configuration, rdfStoreService);
     log.debug("(addTypes) found {} types", types.size());
     types.forEach(type -> addType(configuration, rdfStoreService, shacl, type));
   }
 
-  private List<String> getTypes(RdfStoreService rdfStoreService) {
+  private List<String> getTypes(Configuration configuration, RdfStoreService rdfStoreService) {
     List<Map<String, RDFNode>> rows = getRows(rdfStoreService, "select-types.sparql");
     List<String> types = paginatedQuery.convertSingleColumnUriToStringList(rows);
-    Collections.sort(types);
-    return types;
+
+    Stream<String> prioritisedTypes = types.stream()
+                                    .filter(type -> configuration.isPrioritised(getNamespaceIri(type)));
+    Stream<String> unPrioritisedTypes = types.stream().filter(type -> !configuration.isPrioritised(getNamespaceIri(type)));
+
+    return Stream.concat(
+            prioritisedTypes.sorted(),
+            unPrioritisedTypes.sorted()
+          ).collect(Collectors.toList());
+  }
```

Comment:
@michielrogissart-cognizone should we also not do something similar for properties??